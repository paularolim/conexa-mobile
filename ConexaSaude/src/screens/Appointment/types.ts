import { NativeStackScreenProps } from '@react-navigation/native-stack';

import { Appointment } from '@hooks/useDetailAppointment';
import { MainRoutesParamList } from '@routes/types';

export interface AppointmentViewProps {
  appointment: Appointment | null;
  showModal: boolean;
  handleDismissModal: () => void;
  handleShowCancelModal: () => void;
  handleCancelAppointment: () => void;
}

export type ScreenProps = NativeStackScreenProps<MainRoutesParamList, 'Appointment'>;
