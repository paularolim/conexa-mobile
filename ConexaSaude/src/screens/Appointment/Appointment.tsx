import { View } from 'react-native';

import { Text } from '@components/Text';

export function Appointment() {
  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text>Appointment</Text>
    </View>
  );
}
