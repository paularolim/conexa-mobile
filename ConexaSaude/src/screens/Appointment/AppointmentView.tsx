import { ActivityIndicator, View } from 'react-native';
import { useTheme } from 'styled-components/native';

import { Button } from '@components/Button';
import { Divider } from '@components/Divider';
import { Icon } from '@components/Icon';
import { Modal } from '@components/Modal';

import {
  Avatar,
  Container,
  Content,
  ErrorMessage,
  Footer,
  Header,
  PatientName,
  SectionTitle,
  SectionValue,
} from './styles';
import { AppointmentViewProps } from './types';

export function AppointmentView({
  handleShowCancelModal,
  handleCancelAppointment,
  handleDismissModal,
  showModal,
  appointment,
  error,
  loading,
}: AppointmentViewProps) {
  const { colors } = useTheme();

  if (loading) {
    return (
      <Container>
        <ActivityIndicator size="large" testID="activity-indicator" />
      </Container>
    );
  }

  if (error) {
    return (
      <Container>
        <ErrorMessage testID="error-message">
          Não foi possível carregar os dados. Tente novamente mais tarde!
        </ErrorMessage>
      </Container>
    );
  }

  return (
    <Container>
      <Content>
        <Header>
          <Avatar>
            <Icon name="user" width={80} height={80} fill={colors.pink} />
          </Avatar>
          <PatientName>{appointment?.patient}</PatientName>
        </Header>

        <Divider />

        <View>
          <SectionTitle>Data da consulta</SectionTitle>
          <SectionValue>{appointment?.date}</SectionValue>
        </View>

        <View>
          <SectionTitle>Observações</SectionTitle>
          <SectionValue>{appointment?.observation}</SectionValue>
        </View>
      </Content>

      <Footer>
        <Button>Iniciar consulta</Button>
        <Button type="ghost" onPress={handleShowCancelModal}>
          Cancelar consulta
        </Button>
      </Footer>

      <Modal
        testID="confirm-modal"
        title="Cancelar consulta"
        description="Tem certeza que deseja cancelar a consulta?"
        primaryButtonLabel="Sim, cancelar"
        secondaryButtonLabel="Não, manter consulta"
        visible={showModal}
        dismissModal={handleDismissModal}
        confirm={handleCancelAppointment}
        cancel={handleDismissModal}
      />
    </Container>
  );
}
