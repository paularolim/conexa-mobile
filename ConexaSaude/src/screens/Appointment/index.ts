export * from './Appointment';
export * from './AppointmentView';
export * from './types';
export * from './useAppointmentViewModel';
