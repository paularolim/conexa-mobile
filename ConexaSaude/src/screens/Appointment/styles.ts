import { View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

export const Container = styled(View)`
  flex: 1;
  padding: 24px;
  gap: 16px;
  background-color: ${({ theme }) => theme.colors.white};
`;

export const Content = styled(View)`
  flex: 1;
  gap: 16px;
`;

export const Header = styled(View)`
  flex-direction: row;
  align-items: center;
  gap: 16px;
`;

export const Avatar = styled(View)`
  background-color: ${({ theme }) => theme.colors.grayLight};
  width: 80px;
  height: 80px;
  border-radius: 40px;
  overflow: hidden;
`;

export const PatientName = styled(Text)`
  font-size: 22px;
  font-weight: bold;
`;

export const SectionTitle = styled(Text)`
  font-size: 18px;
  font-weight: bold;
`;

export const SectionValue = styled(Text)`
  font-size: 18px;
`;

export const Footer = styled(View)`
  gap: 4px;
`;

export const ErrorMessage = styled(Text)`
  font-size: 18px;
  color: ${({ theme }) => theme.colors.red};
`;
