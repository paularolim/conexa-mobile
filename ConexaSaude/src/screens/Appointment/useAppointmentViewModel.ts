import { useCallback, useEffect, useState } from 'react';
import { Alert } from 'react-native';

import { useDetailAppointment } from '@hooks/useDetailAppointment';

import { ScreenProps } from './types';

export function useAppointmentViewModel({ route }: ScreenProps) {
  const [showModal, setShowModal] = useState(false);

  const { id } = route.params;

  const { getDetailAppointment, appointment } = useDetailAppointment();

  const requestDetails = useCallback(() => {
    getDetailAppointment(id);
  }, [getDetailAppointment, id]);

  const handleShowCancelModal = () => {
    setShowModal(true);
  };

  const handleCancelAppointment = () => {
    Alert.alert('Cancelada', 'A consulta foi cancelada');
    setShowModal(false);
  };

  const handleDismissModal = () => {
    setShowModal(false);
  };

  useEffect(() => requestDetails(), [requestDetails]);

  return {
    handleShowCancelModal,
    handleCancelAppointment,
    handleDismissModal,
    showModal,
    appointment,
  };
}
