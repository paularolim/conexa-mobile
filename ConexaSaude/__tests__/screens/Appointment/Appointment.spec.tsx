import renderer from 'react-test-renderer';

import { Appointment, NavigationType, RouteType, ScreenProps } from '@screens/Appointment';

import { Wrapper } from '../../Wrapper';

jest.mock('@screens/Appointment/useAppointmentViewModel', () => ({
  useAppointmentViewModel: jest.fn(),
}));

const createTestProps = (): unknown & ScreenProps => ({
  navigation: { navigate: jest.fn() } as unknown as NavigationType,
  route: { params: { id: '1' } } as unknown as RouteType,
});
const props = createTestProps();

describe('AppointmentView', () => {
  it('should render screen correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Appointment navigation={props.navigation} route={props.route} />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
