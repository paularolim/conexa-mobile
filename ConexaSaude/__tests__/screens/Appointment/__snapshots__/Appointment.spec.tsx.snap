// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentView should render screen correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "gap": 16,
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "gap": 16,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#C2CFE0",
              "borderRadius": 40,
              "height": 80,
              "overflow": "hidden",
              "width": 80,
            },
          ]
        }
      >
        <svg
          data-file-name="SvgUser"
          height={80}
          stroke="#F28080"
          width={80}
        />
      </View>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 22,
              "fontWeight": "bold",
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "gap": 10,
          },
        ]
      }
    >
      <View
        primary={true}
        style={
          [
            {
              "backgroundColor": "#F28080",
              "borderRadius": 2,
              "height": 5,
              "width": 80,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#C2CFE0",
              "borderRadius": 2,
              "height": 5,
              "width": 80,
            },
          ]
        }
      />
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Data da consulta
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 18,
            },
          ]
        }
      />
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Observações
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 18,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "gap": 4,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2ED47A",
          "borderRadius": 4,
          "justifyContent": "center",
          "maxWidth": "100%",
          "minHeight": 48,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#FFFFFF",
              "fontSize": 18,
            },
          ]
        }
        type="filled"
      >
        Iniciar consulta
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "justifyContent": "center",
          "maxWidth": "100%",
          "minHeight": 48,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#109CF1",
              "fontSize": 18,
            },
          ]
        }
        type="ghost"
      >
        Cancelar consulta
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    testID="confirm-modal"
    transparent={true}
    visible={true}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#00000066",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "gap": 24,
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "gap": 4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000000",
                },
                {
                  "fontSize": 22,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Cancelar consulta
          </Text>
          <Text
            style={
              [
                {
                  "color": "#000000",
                },
                {
                  "fontSize": 16,
                },
              ]
            }
          >
            Tem certeza que deseja cancelar a consulta?
          </Text>
        </View>
        <View
          style={
            [
              {
                "gap": 4,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#2ED47A",
                "borderRadius": 4,
                "justifyContent": "center",
                "maxWidth": "100%",
                "minHeight": 48,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 8,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                  {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                  },
                ]
              }
              type="filled"
            >
              Sim, cancelar
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "justifyContent": "center",
                "maxWidth": "100%",
                "minHeight": 48,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 8,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                  {
                    "color": "#109CF1",
                    "fontSize": 18,
                  },
                ]
              }
              type="ghost"
            >
              Não, manter consulta
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
